source/common/http/http1/codec_impl.cc

```diff
@@ -575,7 +594,10 @@ StreamEncoder& ClientConnectionImpl::newStream(StreamDecoder& response_decoder)
   if (resetStreamCalled()) {
     throw CodecClientException("cannot create new streams after calling reset");
   }
-
+  // read-enable any underlying connection disabled by flow control on the prior request.
+  while (!connection_.readEnabled()) {
```

Comment:
OK I see what you are saying. I kind of feel like this should be done by the connection pool and not the codec since in some sense this doesn't really have anything to do with the codec. Perhaps here? https://github.com/lyft/envoy/blob/master/source/common/http/http1/conn_pool.cc#L41 (w/ detailed comment added on rationale). WDYT?